rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -2475,6 +2469,17 @@ public Coin getLockingCap() {
         return this.provider.getLockingCap();
     }
 
+    /**
+     * Returns the redeemScript of the current active federation
+     *
+     * @return Returns the redeemScript of the current active federation
+     */
+    public Optional<Script> getActivePowpegRedeemScript() {
+        return activations.isActive(RSKIP293) ?
+            Optional.of(getActiveFederation().getRedeemScript()) :
+            Optional.empty();
```

Comment:
I suggest throwing error or at least logging an error on this kind of unrecoverable situations

Reply:
Do you mean situations where the RSKIP is not active yet?

We are already logging some steps of the call of the method.

![image](https://user-images.githubusercontent.com/94651730/165156272-8a225983-22c5-4e33-969c-82efb87ffa85.png)

What else do you think would be worthy to log?